src/rgw/rgw_lc.cc

```diff
@@ -1489,6 +1440,49 @@ class LCOpAction_Transition : public LCOpAction {
       return 0;
     }
 
+    /* notifications */
+    auto& bucket = oc.bucket;
+    auto& obj = oc.obj;
+
+    RGWObjState* obj_state{nullptr};
+    string etag;
+    r = obj->get_obj_state(oc.dpp, &obj_state, null_yield, true);
+    if (r < 0) {
+      ldpp_dout(oc.dpp, 0) <<
+	fmt::format("ERROR: get_obj_state() failed on transition of object k={} error r={}",
+		    oc.o.key.to_string(), r) << dendl;
+      return r;
+    }
+
+    auto iter = obj_state->attrset.find(RGW_ATTR_ETAG);
+    if (iter != obj_state->attrset.end()) {
+      etag = rgw_bl_str(iter->second);
+    }
+
+    rgw::notify::EventTypeList event_types;
+    event_types.insert(event_types.end(), {rgw::notify::LifecycleTransition});
+    if ((!bucket->versioned()) ||
+	(bucket->versioned() && oc.o.is_current() && !oc.o.is_delete_marker())) {
+      event_types.push_back(rgw::notify::ObjectTransitionCurrent);
+    } else {
+      event_types.push_back(rgw::notify::ObjectTransitionNoncurrent);
+    }
+
+    std::unique_ptr<rgw::sal::Notification> notify =
```

Comment:
This would generate events for all transitions including cloud-transition. Should we remove events from  delete_tier_obj()  (line:1379-1401) to avoid duplicate notifications? (unless we plan to have new notify:: type for deleted objects post cloud-transition)

Reply:
so you would propose to remove all notifications currently being sent in delete_tier_obj()?